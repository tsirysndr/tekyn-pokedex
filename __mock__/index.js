export const pokemon = {
  base_experience: 63,
  height: 6,
  id: 158,
  is_default: true,
  name: "totodile",
  order: 255,
  pokemon_species_id: 158,
  pokemon_v2_pokemonforms: [
    {
      form_name: "",
      form_order: 1,
      id: 158,
      is_battle_only: false,
      is_default: true,
      is_mega: false,
      name: "totodile",
      order: 256,
      pokemon_id: 158,
    },
  ],
  weight: 95,
  pokemon_v2_pokemonabilities: [
    {
      ability_id: 67,
      id: 408,
      is_hidden: false,
      pokemon_id: 158,
      pokemon_v2_ability: {
        name: "torrent",
        pokemon_v2_abilityflavortexts: [
          { id: 4291 },
          { id: 4292 },
          { id: 4293 },
          { id: 4294 },
          { id: 4295 },
          { id: 4296 },
          { id: 4297 },
          { id: 4298 },
          { id: 4299 },
          { id: 4300 },
          { id: 4301 },
          { id: 4302 },
          { id: 4303 },
          { id: 4304 },
          { id: 4305 },
          { id: 4306 },
          { id: 4307 },
          { id: 4308 },
          { id: 4309 },
          { id: 4310 },
          { id: 4311 },
          { id: 4312 },
          { id: 4313 },
          { id: 4314 },
          { id: 4315 },
          { id: 4316 },
          { id: 4317 },
          { id: 4318 },
          { id: 4319 },
          { id: 4320 },
          { id: 4321 },
          { id: 4322 },
          { id: 4323 },
          { id: 4324 },
          { id: 4325 },
          { id: 4326 },
          { id: 4327 },
          { id: 4328 },
          { id: 4329 },
          { id: 4330 },
          { id: 4331 },
          { id: 4332 },
          { id: 4333 },
          { id: 4334 },
          { id: 4335 },
          { id: 4336 },
          { id: 4337 },
          { id: 4338 },
          { id: 4339 },
          { id: 4340 },
          { id: 4341 },
          { id: 4342 },
          { id: 4343 },
          { id: 4344 },
          { id: 4345 },
          { id: 4346 },
          { id: 4347 },
          { id: 4348 },
          { id: 4349 },
          { id: 4350 },
          { id: 4351 },
          { id: 4352 },
          { id: 4353 },
          { id: 4354 },
          { id: 4355 },
        ],
      },
    },
    {
      ability_id: 125,
      id: 409,
      is_hidden: true,
      pokemon_id: 158,
      pokemon_v2_ability: {
        name: "sheer-force",
        pokemon_v2_abilityflavortexts: [
          { id: 7914 },
          { id: 7915 },
          { id: 7916 },
          { id: 7917 },
          { id: 7918 },
          { id: 7919 },
          { id: 7920 },
          { id: 7921 },
          { id: 7922 },
          { id: 7923 },
          { id: 7924 },
          { id: 7925 },
          { id: 7926 },
          { id: 7927 },
          { id: 7928 },
          { id: 7929 },
          { id: 7930 },
          { id: 7931 },
          { id: 7932 },
          { id: 7933 },
          { id: 7934 },
          { id: 7935 },
          { id: 7936 },
          { id: 7937 },
          { id: 7938 },
          { id: 7939 },
          { id: 7940 },
          { id: 7941 },
          { id: 7942 },
          { id: 7943 },
          { id: 7944 },
          { id: 7945 },
          { id: 7946 },
          { id: 7947 },
          { id: 7948 },
          { id: 7949 },
          { id: 7950 },
          { id: 7951 },
          { id: 7952 },
          { id: 7953 },
          { id: 7954 },
          { id: 7955 },
          { id: 7956 },
          { id: 7957 },
          { id: 7958 },
          { id: 7959 },
          { id: 7960 },
          { id: 7961 },
          { id: 7962 },
          { id: 7963 },
          { id: 7964 },
          { id: 7965 },
          { id: 7966 },
          { id: 7967 },
          { id: 7968 },
          { id: 7969 },
          { id: 7970 },
          { id: 7971 },
          { id: 7972 },
        ],
      },
    },
  ],
  pokemon_v2_pokemonstats: [
    {
      base_stat: 50,
      effort: 0,
      id: 943,
      pokemon_id: 158,
      stat_id: 1,
      pokemon_v2_stat: { name: "hp" },
    },
    {
      base_stat: 65,
      effort: 1,
      id: 944,
      pokemon_id: 158,
      stat_id: 2,
      pokemon_v2_stat: { name: "attack" },
    },
    {
      base_stat: 64,
      effort: 0,
      id: 945,
      pokemon_id: 158,
      stat_id: 3,
      pokemon_v2_stat: { name: "defense" },
    },
    {
      base_stat: 44,
      effort: 0,
      id: 946,
      pokemon_id: 158,
      stat_id: 4,
      pokemon_v2_stat: {
        name: "special-attack",
      },
    },
    {
      base_stat: 48,
      effort: 0,
      id: 947,
      pokemon_id: 158,
      stat_id: 5,
      pokemon_v2_stat: {
        name: "special-defense",
      },
    },
    {
      base_stat: 43,
      effort: 0,
      id: 948,
      pokemon_id: 158,
      stat_id: 6,
      pokemon_v2_stat: { name: "speed" },
    },
  ],
  pokemon_v2_pokemontypes: [
    {
      pokemon_id: 158,
      pokemon_v2_type: {
        name: "water",
        pokemon_v2_generation: {
          name: "generation-i",
        },
      },
    },
  ],
};

export const pokemonspecies = [
  {
    id: 158,
    pokemon_v2_pokemons: [{ name: "totodile" }],
    pokemon_v2_pokemonspeciesflavortexts: [
      {
        language_id: 9,
        flavor_text:
          "Its well-developed\njaws are powerful\nand capable of\fcrushing anything.\nEven its trainer\nmust be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It is small but\nrough and tough.\nIt won't hesitate\fto take a bite out\nof anything that\nmoves.",
      },
      {
        language_id: 9,
        flavor_text:
          "This rough critter\nchomps at any\nmoving object it\fsees. Turning your\nback on it is not\nrecommended.",
      },
      {
        language_id: 9,
        flavor_text:
          "Despite the smallness of its body,\nTOTODILE’s jaws are very powerful.\nWhile the POKéMON may think it is just\fplayfully nipping, its bite has enough\npower to cause serious injury.",
      },
      {
        language_id: 9,
        flavor_text:
          "Despite the smallness of its body,\nTOTODILE’s jaws are very powerful.\nWhile the POKéMON may think it is just\fplayfully nipping, its bite has enough\npower to cause serious injury.",
      },
      {
        language_id: 9,
        flavor_text:
          "Despite its small body, TOTODILE’s jaws\nare very powerful. While it may think it is\njust playfully nipping, its bite has enough\nstrength to cause serious injury.",
      },
      {
        language_id: 9,
        flavor_text:
          "It is small but rough and tough. It won’t\nhesitate to take a bite out of anything\nthat moves.",
      },
      {
        language_id: 9,
        flavor_text:
          "Its well-developed jaws are powerful and\ncapable of crushing anything. Even its\nTRAINER must be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It has the habit of biting anything\nwith its developed jaws. Even its\nTrainer needs to be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It has the habit of biting anything\nwith its developed jaws. Even its\nTrainer needs to be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It has the habit of biting anything\nwith its developed jaws. Even its\nTrainer needs to be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "Its powerful, well-developed jaws\nare capable of crushing anything.\nEven its Trainer must be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It is small but rough and tough.\nIt won’t hesitate to take a bite out\nof anything that moves.",
      },
      {
        language_id: 9,
        flavor_text:
          "It has the habit of biting anything\nwith its developed jaws. Even its\nTrainer needs to be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It has the habit of biting anything\nwith its developed jaws. Even its\nTrainer needs to be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It has the habit of biting anything\nwith its developed jaws. Even its\nTrainer needs to be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It has the habit of biting anything\nwith its developed jaws. Even its\nTrainer needs to be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "It is small but rough and tough. It won’t hesitate to\ntake a bite out of anything that moves.",
      },
      {
        language_id: 9,
        flavor_text:
          "Its powerful, well-developed jaws are capable of\ncrushing anything. Even its Trainer must be careful.",
      },
      {
        language_id: 9,
        flavor_text:
          "Despite the smallness of its body, Totodile’s jaws are very\npowerful. While the Pokémon may think it is just playfully\nnipping, its bite has enough power to cause serious injury.",
      },
      {
        language_id: 9,
        flavor_text:
          "Despite the smallness of its body, Totodile’s jaws are very\npowerful. While the Pokémon may think it is just playfully\nnipping, its bite has enough power to cause serious injury.",
      },
    ],
  },
  {
    id: 159,
    pokemon_v2_pokemons: [{ name: "croconaw" }],
    pokemon_v2_pokemonspeciesflavortexts: [],
  },
  {
    id: 160,
    pokemon_v2_pokemons: [{ name: "feraligatr" }],
    pokemon_v2_pokemonspeciesflavortexts: [],
  },
];

export const pokemonstats = [
  {
    base_stat: 50,
    effort: 0,
    id: 943,
    pokemon_id: 158,
    stat_id: 1,
    pokemon_v2_stat: { name: "hp" },
  },
  {
    base_stat: 65,
    effort: 1,
    id: 944,
    pokemon_id: 158,
    stat_id: 2,
    pokemon_v2_stat: { name: "attack" },
  },
  {
    base_stat: 64,
    effort: 0,
    id: 945,
    pokemon_id: 158,
    stat_id: 3,
    pokemon_v2_stat: { name: "defense" },
  },
  {
    base_stat: 44,
    effort: 0,
    id: 946,
    pokemon_id: 158,
    stat_id: 4,
    pokemon_v2_stat: { name: "special-attack" },
  },
  {
    base_stat: 48,
    effort: 0,
    id: 947,
    pokemon_id: 158,
    stat_id: 5,
    pokemon_v2_stat: { name: "special-defense" },
  },
  {
    base_stat: 43,
    effort: 0,
    id: 948,
    pokemon_id: 158,
    stat_id: 6,
    pokemon_v2_stat: { name: "speed" },
  },
];
