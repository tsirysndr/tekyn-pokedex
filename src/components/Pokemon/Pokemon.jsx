import React from "react";
import PropTypes from "prop-types";
import { Image, View } from "react-native";
import DefaultText from "@Components/Text";
import { useFonts, Roboto_700Bold } from "@expo-google-fonts/roboto";
import styled, { css } from "@emotion/native";
import zeropad from "zeropad";
import { POKEMON_IMAGE_URL } from "src/constants/urls";

const Container = styled.View`
  width: 100px;
  justify-content: center;
  align-items: center;
`;

const Text = styled(DefaultText)`
  text-align: center;
  ${({ fontsLoaded }) =>
    fontsLoaded &&
    css`
      font-family: "Roboto_700Bold";
    `}
`;

export default function Pokemon({ id, name, fontSize, size }) {
  const [fontsLoaded] = useFonts({ Roboto_700Bold });
  const format = (value) => `#${zeropad(value, 3)}`;
  return (
    <Container>
      <Image
        style={{ height: size, width: size }}
        resizeMode="cover"
        source={{ uri: `${POKEMON_IMAGE_URL}/${zeropad(id, 3)}.png` }}
      />
      <Text fontsLoaded={fontsLoaded} color="#264653" style={{ fontSize }}>
        {format(id)}
      </Text>
      <Text fontsLoaded={fontsLoaded} color="#264653" style={{ fontSize }}>
        {name}
      </Text>
    </Container>
  );
}

Pokemon.defaultProps = {
  fontSize: 10,
  size: 56,
};

Pokemon.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  fontSize: PropTypes.number,
  size: PropTypes.number,
};
