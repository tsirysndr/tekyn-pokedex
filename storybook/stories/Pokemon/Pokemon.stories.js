import React from "react";
import Pokemon from "@Components/Pokemon";
import { storiesOf } from "@storybook/react-native";
import CenterView from "../CenterView";

storiesOf("Pokemon", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("Charmander", () => <Pokemon id={4} name='Charmander' />)
  .add("Charmeleon", () => <Pokemon id={5} name='Charmeleon' />)
  .add("Charizard", () => <Pokemon id={6} name='Charizard' />)
  .add("Drowzee", () => <Pokemon id={96} name='Drowzee' />)
  .add("Hypno", () => <Pokemon id={97} name='Hypno' />);