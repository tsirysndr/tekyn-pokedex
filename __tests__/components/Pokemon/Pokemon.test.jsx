import Pokemon from "@Components/Pokemon";
import { render } from "@testing-library/react-native";
import { pokemon, pokemonspecies } from "__mock__";

test("renders Pokemon", () => {
  const tree = render(
    <Pokemon />
  );
  expect(tree).toMatchSnapshot();
});
