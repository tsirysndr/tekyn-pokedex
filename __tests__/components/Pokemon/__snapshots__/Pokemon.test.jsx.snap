// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pokemon 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 100,
      },
      undefined,
    ]
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/025.png",
      }
    }
    style={
      Object {
        "height": 56,
        "width": 56,
      }
    }
  />
  <Text
    color="#264653"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#264653",
        },
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 10,
          },
        ],
      ]
    }
  >
    #025
  </Text>
  <Text
    color="#264653"
    fontsLoaded={false}
    style={
      Array [
        Object {
          "color": "#264653",
        },
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 10,
          },
        ],
      ]
    }
  >
    Pikachu
  </Text>
</View>
`;
